src/crowdsale/CrowdSaleWithFees.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { CrowdSale, Sale, SaleInfo, AlreadyClaimed, SaleState, BadSaleState } from "./CrowdSale.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title CrowdSaleWithFees
+ * @author molecule.to
+ * @notice a plain crowdsale that takes 0.5% fees of the funding goal upon sale settlement
+ */
+contract CrowdSaleWithFees is CrowdSale, Ownable {
+    using SafeERC20 for IERC20Metadata;
+
+    mapping(uint256 => uint256) crowdSaleFees;
+    uint256 public feesPercentage;
```

Comment:
the "percentage" is a value < 10_000 and hence should nicely fit into an uint16.